src/rpc/WorkQueue.cpp

```diff
@@ -19,13 +19,40 @@
 
 #include "rpc/WorkQueue.hpp"
 
+#include "util/config/Config.hpp"
+#include "util/log/Logger.hpp"
 #include "util/prometheus/Label.hpp"
 #include "util/prometheus/Prometheus.hpp"
 
+#include <boost/json/object.hpp>
+
+#include <cstddef>
 #include <cstdint>
+#include <functional>
+#include <thread>
+#include <utility>
 
 namespace rpc {
 
+void
+WorkQueue::OneTimeCallable::setCallable(std::function<void()> func)
+{
+    func_ = func;
+}
+
+void
+WorkQueue::OneTimeCallable::operator()()
+{
+    if (not called_) {
+        func_();
```

Comment:
I feel there is still race condition here. Can worker thread and stop thread both be running func_ ?

Reply:
It is inside `Mutex`, so this can't be called from multiple threads.